packages/@aws-cdk/aws-lambda/lib/code-signing-config.ts

```diff
@@ -0,0 +1,128 @@
+   * @attribute CodeSigningConfigId
+   */
+  readonly codeSigningConfigId: string;
+}
+
+/**
+ * Construction properties for a Code Signing Config object
+ */
+export interface CodeSigningConfigProps {
+  /**
+   * List of signing profiles that defines a
+   * trusted user who can sign a code package.
+   */
+  readonly signingProfiles: ISigningProfile[],
+
+  /**
+   * Code signing configuration policy for deployment validation failure.
+   * If you set the policy to Enforce, Lambda blocks the deployment request
+   * if signature validation checks fail.
+   * If you set the policy to Warn, Lambda allows the deployment and
+   * creates a CloudWatch log.
+   *
+   * @default - UntrustedArtifactOnDeployment.WARN
+   */
+  readonly untrustedArtifactOnDeployment?: UntrustedArtifactOnDeployment,
+
+  /**
+   * Code signing configuration description.
+   *
+   * @default - No description.
+   */
+  readonly description?: string
+
+  /**
+   * Physical name of this Code Signing Config.
+   *
+   * @default - Assigned by CloudFormation (recommended).
+   */
+  readonly codeSigningConfigName?: string;
+}
+
+/**
+ * Defines a Code Signing Config.
+ *
+ * @resource AWS::Lambda::CodeSigningConfig
+ */
+export class CodeSigningConfig extends Resource implements ICodeSigningConfig {
+  /**
+   * Creates a Signing Profile construct that represents an external Signing Profile.
+   *
+   * @param scope The parent creating construct (usually `this`).
+   * @param id The construct's name.
+   * @param codeSigningConfigArn The ARN of code signing config.
+   */
+  public static fromCodeSigningConfigArn( scope: Construct, id: string, codeSigningConfigArn: string): ICodeSigningConfig {
+    class Import extends Resource implements ICodeSigningConfig {
+      public readonly codeSigningConfigArn = codeSigningConfigArn;
+      public readonly codeSigningConfigId: string;
+
+      constructor( codeSigningProfileId: string ) {
+        super(scope, id);
+        this.codeSigningConfigId = codeSigningProfileId;
+      }
+    }
+    const codeSigningProfileId = Stack.of(scope).parseArn(codeSigningConfigArn).resourceName;
+    if (!codeSigningProfileId) {
+      throw new Error(`Code signing config ARN must be in the format 'arn:aws:lambda:<region>:<account>:code-signing-config:<codeSigningConfigArn>', got: '${codeSigningConfigArn}'`);
+    }
+    return new Import(codeSigningProfileId);
+  }
+
+  readonly codeSigningConfigArn: string;
+  readonly codeSigningConfigId: string;
+
+  constructor(scope: Construct, id: string, props: CodeSigningConfigProps) {
+    super(scope, id, {
+      physicalName: props.codeSigningConfigName,
+    });
+
+    const signingProfileVersionArns = props.signingProfiles.map(signingProfile => {
+      return signingProfile.signingProfileProfileVersionArn;
+    });
+
+    const resource: CfnCodeSigningConfig = new CfnCodeSigningConfig(this, 'Resource', {
+      allowedPublishers: {
+        signingProfileVersionArns,
+      },
```

Comment:
What is a signing profile version and how is it different from a signing profile? When is a new version generated?

It's important to understand to learn about how much control we should give to the user in choosing an older version, and to also decide what the correct default behaviour should be.

In the current code, it looks like, if the signing profile is updated to produce a new version, that new version will be used here. It's not clear if that's the correct behaviour we should add to the CDK.

Reply:
@nija-at 
I'm sorry but, I could not find any documents that show the timing to generate new version of signing profile.
It seems to version is generated when signing profile generated and never changed. I’m not sure the use case of version.
Do you have any ideas on how to proceed with the implementation?

Reply:
I will take a look at this early next week, and let you know how we can move forward. Sorry for the delay.

Reply:
[This doc](https://docs.aws.amazon.com/signer/latest/developerguide/gs-profile.html) states - 
> If code signing is enabled for Lambda functions, only zip files signed by specified signing profile versions will pass signature validation checks.

This means that lambda code packages must be signed by the exact signing profile version. 

Comparing this to the [SigningProfile CloudFormation spec](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-signer-signingprofile.html), a new version is generated only when the platform id is changed and signature validity period is modified.

At this point, I think what you have is fine. Somehow the user will need to coordinate updating the signing profile and enforcing it. As a later change, we'll need to consider making this better.

Reply:
@nija-at 
Thank you for doing the research! 
I understand that 
'we can only replace the signing profile currently, but if we can update the signing profile in the future, we should follow suit then.'

So, is there any other code that I needs to modify?

I'm sorry to ask this, but I'd like to correct a mistake in the author setting for a previous commit.
May I rebase it like this? 
https://github.com/hedrall/aws-cdk/commits/aws-lambda-code-signing-rebased